### ISSUE TITLE
[Bug] Dynamic modelling for water heat exchanger does not work

### ISSUE THREAD

**Issue by WYuanJun** [DanWBR/dwsim6#429]
IMPORTANT: This section is for code-related bugs. For usage issues, please go to the [Support Forums](https://sourceforge.net/p/dwsim/discussion/).

**Describe the bug**
When trying to solve a water to water heat exchanger in dynamic mode the pressure and temperature goes out of range. The model solves in steady state mode. 
It then has an error: the given key was not present in dictionary.

Note: The thermodynamic property package used is Raoult's Law. (The same problem happens in IAPWS steam table package)

**To Reproduce**
1. Created a water to water heat exchangers hot inlet at 53°C and the cold inlet at 26°C with PID control on the cold stream outlet valve and specification block for the hot stream mass flow rate.
2. Click solve in steady state mode. It solves. 
3. Click on dynamic mode to activate dynamic mode, then run schedule.
4. See the information panel for error.

**Expected behavior**
I would expect the heat exchanger to find a valve opening percentage with the correct pressure and flow in dynamic mode.

**Screenshots**
![image](https://user-images.githubusercontent.com/125230606/218355900-19ca9981-e830-41d2-a2f5-0735c890c96f.png)
![image](https://user-images.githubusercontent.com/125230606/218355935-7571d85b-4e62-4a86-aa88-7c8516c826dd.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version v8.3.5

**Additional context**
The heat exchanger calculation type is calculate outlet temperatures (UA) mode.
This is in normal DWSIM not the Cross Platform

Many thanks

**Comment by DanWBR** [comment]
@WYuanJun can you share the flowsheet?

**Comment by WYuanJun** [comment]
> @WYuanJun can you share the flowsheet?

@DanWBR Hi Dan, this comment section does not seem to let me upload a DWSIM file. 
Can you please advise me how can I best share with you the flowsheet? Many thanks!

**Comment by WYuanJun** [comment]
> @WYuanJun can you share the flowsheet?

Hi Dan,

The testHX6 file is the original with the problem but the "given key not present" error has somehow gone away but the model still does not show the correct temperature, pressure and mass flow as I would expect.

The testHX7 file is also another model with the exact same process model I tried to recreate but with different heat exchanger parameter, UA and area but I am still not getting the correct output for temperature, pressure and flow rates.

Your support and advice would be great appreciated! Thank you!
[testHX6.zip](https://github.com/DanWBR/dwsim/files/10775454/testHX6.zip)

[testHX7.zip](https://github.com/DanWBR/dwsim/files/10775455/testHX7.zip)

**Comment by DanWBR** [comment]
@WYuanJun something is missing here: 
![image](https://user-images.githubusercontent.com/2669419/220132819-78f796ce-10db-45a0-99b8-e3e96e969fb4.png)

as for HX7, it doesn't look right indeed. I'll take a look

**Comment by WYuanJun** [comment]
> @WYuanJun something is missing here: ![image](https://user-images.githubusercontent.com/2669419/220132819-78f796ce-10db-45a0-99b8-e3e96e969fb4.png)
> 
> as for HX7, it doesn't look right indeed. I'll take a look

Oh yes for the screenshot I had the other outlet streams from the heat exchanger before and it had the same problem as HX7 so I deleted those streams (it was named 8 & 9) so that is why it is missing from the monitored variables. Sorry I forgot I did that. Maybe that is what is causing the problem. 

But yes HX7 still has some issues I believe

**Comment by DanWBR** [comment]
@WYuanJun what exactly are you trying to control? you selected mass flow but inserted a value like 1 bar, this doesn't make sense.

**Comment by DanWBR** [comment]
@WYuanJun it doesn't make sense to control pressure if you have only liquid in the process.

**Comment by WYuanJun** [comment]
Hi Dan

You are right I should be using flow in the dynamics tab to control the flow.

I guess I wasn't thinking it through at the time, but I was just trying to copy the heat exchanger example but with just water instead. 

For HX7 I was trying to control the outlet (stream 10) temperature to be at 50C by changing the opening of valve 3 to do so to change the flow rate to control the outlet temperature. 

Sorry I am not very experienced with dynamic modelling and are trying to learn. 
I have attached the HX7_v3 and in there I have changed the dynamics from pressure to flow. But again, the pressure and temperature are still out of range.

Many thanks!
[testHX7_v3.zip](https://github.com/DanWBR/dwsim/files/10808143/testHX7_v3.zip)


Cheers
